util/topgen/templates/toplevel_memory.ld.tpl

```diff
@@ -17,6 +18,21 @@ def memory_to_flags(memory):
         flags_str += "x"
 
     return flags_str
+
+def flags(mem):
+    swaccess = mem["swaccess"]
+    exec = mem["exec"]
+    assert swaccess in ["ro", "rw"]
+    flags_str = ""
+    if swaccess == "ro":
+        flags_str += "r"
+    else:
+        flags_str += "rw"
```

Comment:
Maybe this would be clearer with a dictionary lookup?
```python3
    sw_to_flags = {
        'rw': 'rw'
        'ro': 'r'
    }
    assert swaccess in sw_to_flags
    flags_str = sw_to_flags[swaccess]
```
Same number of lines, but the mapping is a bit more obvious.

Reply:
ok let me change that